Assets/Scripts/Spawner/api/CreatureSpawnAction.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Creatures.Api;
+
+namespace Spawner.Api
+{
+    /*
+     * Base class for spawning creatures, distinguished this as 
+     * creature spawner in order to leave room for spawners that
+     * can spawn things like treasure, objectives, events, etc.
+     */
+    public abstract class CreatureSpawnAction
```

Comment:
Wonder if c# has functional interfaces...

Reply:
Think we would use these: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/delegates/